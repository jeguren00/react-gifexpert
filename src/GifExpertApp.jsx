import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);
    const onAddCategory = (value) => {
        if (categories.includes(value)) return;
        const newCategories = [value,...categories];
        setCategories(newCategories);
    };

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory onAddCategory={onAddCategory}/>
            <ol>
                { categories.map( category => {
                    return <GifGrid key={category} category={category}/>
                })}
            </ol>
        </>
    );
}