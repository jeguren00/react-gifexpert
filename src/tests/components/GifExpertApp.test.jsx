import { fireEvent, render,screen } from "@testing-library/react"
import { GifExpertApp, onAddCategory } from "../../GifExpertApp";


describe('pruebas en GifExpertApp', () => {
    const inputValue = 'fallout';
    test('to match snapshot', () => { 
        const {container} = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
     });

     test('aÃ±adir una categoria al input', () => { 
        render(<GifExpertApp />);
        const input = screen.getByRole('textbox');
        fireEvent.input(input, {target: {value: inputValue}});
        expect(input.value).toBe(inputValue);    
    });

});