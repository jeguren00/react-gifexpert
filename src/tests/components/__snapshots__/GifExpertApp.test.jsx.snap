// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifExpertApp to match snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <ol>
    <h3>
      One Punch
    </h3>
    <h2>
      Cargando....
    </h2>
    <div
      class="card-grid"
    />
    <h3>
      Dragon Ball
    </h3>
    <h2>
      Cargando....
    </h2>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
